octodns/provider/powerdns.py

```diff
@@ -381,6 +382,12 @@ def _apply(self, plan):
         for change in changes:
             class_name = change.__class__.__name__
             mods.append(getattr(self, '_mod_{}'.format(class_name))(change))
+
+        # Ensure that any DELETE modifications always occur before any REPLACE
+        # modifications. This ensures that an A record can be replaced by a
+        # CNAME record and vice-versa.
+        mods = sorted(mods, key=itemgetter('changetype'))
```

Comment:
I'm relying on the fact that PowerDNS only has two valid values for `changetype`, being `DELETE` or `REPLACE`, and `DELETE` is ahead of `REPLACE` alphabetically.

Ref: https://doc.powerdns.com/authoritative/http-api/zone.html#rrset